client/my-sites/stats/stats-purchase/use-available-upgrade-tiers.tsx

```diff
@@ -61,26 +65,54 @@ function transformTier( tier: PriceTierListItemProps ): StatsPlanTierUI {
 	};
 }
 
-function getPlanTiersForSite( siteId: number | null, tiers: StatsPlanTierUI[] ): StatsPlanTierUI[] {
-	// TODO: Determine if we need to filter tiers locally.
-	// Accept the fill list of tiers and filter out options
-	// that don't apply to the current site (ie: only upgrades, not downgrades)
-	// Could happen on the server, in which case, we could remove this step.
+function filterPurchasedTiers(
+	availableTiers: StatsPlanTierUI[],
+	purchasedTiers: PlanUsage | undefined
+): StatsPlanTierUI[] {
+	// Filter out already purchased tiers.
+	let tiers: StatsPlanTierUI[];
+
+	if (
+		! purchasedTiers ||
+		purchasedTiers?.views_limit === null ||
+		purchasedTiers?.views_limit === 0
+	) {
+		// No tier has been purchased.
+		tiers = availableTiers;
+	} else {
+		tiers = availableTiers.filter( ( availableTier ) => {
+			if ( ! availableTier?.views || ! purchasedTiers?.views_limit ) {
```

Comment:
Is the condition `purchasedTiers?.views_limit` necessary since there has been a [similar check above](https://github.com/Automattic/wp-calypso/pull/84847/files#diff-37d48649a618d0e3e046ddaeabed44e96ae0ed93a49ec01b52a04d4dd7da0a02R77-R78?)?

Reply:
Is the condition `! availableTier?.views` the same as the check below?

Reply:
Thanks @dognose24. I see now that I can remove this check. 
In the beginning, TS complained about undefined values, but this condition became unnecessary after other changes.